presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -0,0 +1,451 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.metadata.FunctionExtractor.extractFunctions;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+
+public class TestGeoQueries
+        extends AbstractTestQueryFramework
+{
+    public TestGeoQueries()
+    {
+        super(TestGeoQueries::createLocalQueryRunner);
+    }
+
+    @Test
+    public void testSTPoint()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Point(1, 4))", "select 'POINT (1 4)'");
+        assertQuery("select ST_AsText(ST_Point(122.3, 10.55))", "select 'POINT (122.3 10.55)'");
+    }
+
+    @Test
+    public void testSTLineFromText()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_LineFromText('LINESTRING EMPTY'))", "select 'MULTILINESTRING EMPTY'");
+        assertQuery("select ST_AsText(ST_LineFromText('LineString(1 1, 2 2, 1 3)'))", "select 'LINESTRING (1 1, 2 2, 1 3)'");
+        assertQueryFails("select ST_AsText(ST_LineFromText('MULTILineString EMPTY'))", "ST_LineFromText only applies to LineString. Input type is: MultiLineString");
+        assertQueryFails("select ST_AsText(ST_LineFromText('Polygon ((1 1, 1 4, 4 4, 4 1))'))", "ST_LineFromText only applies to LineString. Input type is: Polygon");
+    }
+
+    @Test
+    public void testSTPolygon()
+            throws Exception
+    {
+        assertQuery("select ST_AsText(ST_Polygon('POLYGON EMPTY'))", "select 'MULTIPOLYGON EMPTY'");
+        assertQuery("select ST_AsText(ST_Polygon('Polygon ((1 1, 1 4, 4 4, 4 1))'))", "select 'POLYGON ((1 1, 4 1, 4 4, 1 4, 1 1))'");
+        assertQueryFails("select ST_AsText(ST_Polygon('LineString(1 1, 2 2, 1 3)'))", "ST_Polygon only applies to Polygon. Input type is: LineString");
+    }
+
+    @Test
+    public void testSTArea()
+            throws Exception
+    {
+        assertQuery("select ST_Area(ST_GeometryFromText('Polygon ((2 2, 2 6, 6 6, 6 2))'))", "select 16.0");
+        assertQueryFails("select ST_Area(ST_GeometryFromText('Point (1 4)'))", "ST_Area only applies to Polygon. Input type is: Point");
+        assertQueryFails("select ST_Area(ST_GeometryFromText('POLYGON EMPTY'))", "ST_Area only applies to Polygon. Input type is: MultiPolygon");
```

Comment:
ST_Area for an empty polygon should not fail. According to SQL/MM spec it should return null. Also, I noticed that ST_Length for empty line-string returns 0. Let's make these two method consistent: both return 0 or null.

8.1.2 ST_Area Methods ...If SELF is an empty set, then return the null value.

BTW, why does ST_Area not allow multi-polygons?